stat/roc.go

```diff
@@ -126,3 +128,67 @@ func ROC(cutoffs, y []float64, classes []bool, weights []float64) (tpr, fpr, thr
 
 	return tpr, fpr, cutoffs
 }
+
+// TOC returns the Total Operating Characteristic for the classes provided
+// and the minimum and maximum bounds for the TOC.
+//
+// The input classes should be sorted with their weights according to the
+// rank variable ascending. SortWeightedLabeled can be used to sort classes
+// together with weights by the rank variable.
+//
+// The returned ntp values can be interpreted as true positives where values
+// at or above the given rank are assigned class true for each given rank
+// from 1 to len(classes). The values of min and max provide the minimum and
+// maximum possible number of true values for the set of classes.
+//
+// If weights is nil, all weights are treated as 1. If weights is not nil
+// it must have the same length as classes, otherwise TOC will panic.
+//
+// More details about TOC curves are available at
+// https://en.wikipedia.org/wiki/Total_operating_characteristic
+func TOC(classes []bool, weights []float64) (min, ntp, max []float64) {
+	if weights != nil && len(classes) != len(weights) {
+		panic("stat: slice length mismatch")
+	}
+	if len(classes) == 0 {
+		return nil, nil, nil
+	}
+
+	ntp = make([]float64, len(classes)+1)
+	min = make([]float64, len(ntp))
+	max = make([]float64, len(ntp))
+	if weights == nil {
+		for i := range ntp[1:] {
+			ntp[i+1] = ntp[i]
+			if !classes[i] {
```

Comment:
When you implemented the "parity switch" you changed this `classes[i]` to `!classes[i]`, I don't think this is correct --- it should still be `classes[i]`, otherwise this is no longer calculating true positives at all. The confusion is to do with the ranking variable vs the ranks themselves as I discuss in the other thread.

Reply:
This matches the changes in [#959](https://github.com/gonum/gonum/pull/959/files#diff-07d8944b2b9171f3dfbb983bfc016c30c93b829e6d2b2c4c91b21de24f3c8dccR96). If this is wrong, so is that. This was part of the reason for making the AUC examples for the TOC and extending the AUC examples to cover the weighted case; they must match.

Reply:
Hmm, I need to think this through again from the beggining...

Reply:
The docstring _here_ says "where values at or above the given rank are assigned class true". The discussion at #316 says "ROC treats values below the cutoff as true (rather than false)", leading me to think that the docstring here is out of sync with the code.

Reply:
Yes, this reflects the doc comment for ROC, which I neglected to invert when I inverted the logic. That should be fixed as well (separate PR). Do you agree?

Reply:
I disagree. I think the docstring for the new function should be fixed now, to avoid submitting a description which we know is incorrect just for symmetry's sake. The ROC comment can be fixed in another PR.

Reply:
To explain more clearly, the ROC doc comment should be fixed in another PR. I'm still not convinced that this PR should go into v0.9, but a fix for the ROC docs should. An inversion of the TOC doc comment should go in this PR.

Reply:
I agree :)

Reply:
OK, I think that's fixed here too.